modules/vmseries/variables.tf

```diff
@@ -120,9 +120,33 @@ variable "img_version" {
   type        = string
 }
 
-variable "name_application_insights" {
-  default     = null
+variable "create_application_insights" {
+  description = "Enable usage of the Application Insights. The default vailue is set to 'true'."
+  default     = true
+  type        = bool
+}
+
+variable "application_insights_name" {
   description = "Name of the Applications Insights instance to be created. Can be `null`, in which case a default name is auto-generated."
+  default     = null
+  type        = string
+}
+
+variable "application_insights_workspace_mode" {
+  description = "Set Application Insights mode to \"Workspace-based\". If set to `false`, the legacy \"Classic\" mode is used. "
+  default     = true
+  type        = bool
+}
+
+variable "log_analytics_workspace_name" {
+  description = "Name of the Log Analytics workspace to be created. Can be `null`, in which case a default name is auto-generated."
+  default     = null
+  type        = string
+}
+
+variable "log_analytics_workspace_sku" {
+  description = "Azure Log Analytics Workspace mode SKU. The default value is set to \"PerGB2018\""
```

Comment:
I think a good idea would be to point to MS documentation, to a list of possible values.

Reply:
Agreed. Also, no need for the last sentence, since it's already stated in the line below :)